src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -763,44 +763,52 @@ public SocketError ReceiveFrom(byte[] buffer, int offset, int count, ref SocketF
         {
             Debug.Assert(timeout == -1 || timeout > 0, $"Unexpected timeout: {timeout}");
 
-            SocketFlags receivedFlags;
-            SocketError errorCode;
-            if (SocketPal.TryCompleteReceiveFrom(_socket, buffer, offset, count, flags, socketAddress, ref socketAddressLen, out bytesReceived, out receivedFlags, out errorCode))
-            {
-                flags = receivedFlags;
-                return errorCode;
-            }
-
-            using (var @event = new ManualResetEventSlim(false, 0))
+            ManualResetEventSlim @event = null;
```

Comment:
FWIW, given this usage of MRES, Dispose'ing it is a nop, so it may not even be worth the extra code complexity to have a try/finally.  Plus, at some point, we might want to look into pooling these.